File: src/main/java/com/study/dong/member/controller/MemberController.java
package com.study.dong.member.controller;


import com.study.dong.member.dto.MemberDto;
import com.study.dong.member.service.MemberService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;


@RequiredArgsConstructor
@Controller
public class MemberController {
    private final MemberService memberService;

    @GetMapping("/member")
    public String asd(){
        return "/member/index";
    }

    @GetMapping("/signup")
    public String signupForm(Model model){
        model.addAttribute("member", new MemberDto());

        return "/member/signup";
    }

    @PostMapping("/signup")
    public String signup(MemberDto memberDto){
        memberService.signUp(memberDto);

        return "redirect:/";
    }

    @GetMapping("/login")
    public String login(){
        return "/member/login";
    }
    
    @GetMapping("/member/info")
    @ResponseBody
    public String info(){
        return "Test";
    }
}

File: 스프링 부트 핵심 가이드/src/main/kotlin/dong/kotlin_study/controller/MemberController.kt
package dong.kotlin_study.controller

import dong.kotlin_study.domain.Member
import dong.kotlin_study.dto.MemberForm
import dong.kotlin_study.service.MemberService
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping

@Controller
class MemberController(val memberService: MemberService) {

    @GetMapping("/members/new")
    fun createForm(): String = "members/createMemberForm"

    @PostMapping("/members/new")
    fun create(form: MemberForm): String {
        val member = Member(name = form.name)

        memberService.join(member);

        return "redirect:/"
    }

    @GetMapping("/members")
    fun list(model: Model): String {
        val members = memberService.findMembers()
        model.addAttribute("members", members)
        return "members/memberList"
    }
}